manifests/init.pp

```diff
@@ -64,118 +79,185 @@
-    yum_max_speed          => $pulp_max_speed,
-    manage_broker          => false,
-    manage_httpd           => false,
-    manage_plugins_httpd   => true,
-    manage_squid           => true,
-    enable_rpm             => true,
-    enable_puppet          => true,
-    enable_docker          => true,
-    enable_ostree          => $enable_ostree,
-    num_workers            => $num_pulp_workers,
-    max_tasks_per_child    => $max_tasks_per_pulp_worker,
-    enable_parent_node     => false,
-    repo_auth              => true,
-    puppet_wsgi_processes  => 1,
-    crane_data_dir         => '/var/lib/pulp/published/docker/v2/app',
-    enable_katello         => true,
-  } ~>
-  class { '::qpid::client':
-    ssl                    => true,
-    ssl_cert_name          => 'broker',
-    ssl_cert_db            => $certs::nss_db_dir,
-    ssl_cert_password_file => $certs::qpid::nss_db_password_file,
-  } ~>
-  class { '::katello::qpid':
-    client_cert => $certs::qpid::client_cert,
-    client_key  => $certs::qpid::client_key,
+  # The ordering of Class['certs'] before Class['certs::*'] should be in puppet-certs, will remove once its handled there
+  if $enable_candlepin {
+    Class['certs'] ~>
+    class { '::certs::candlepin': } ~>
+    class { '::candlepin':
+      user_groups                  => $katello::user_groups,
+      oauth_key                    => $katello::oauth_key,
+      oauth_secret                 => $katello::oauth_secret,
+      deployment_url               => $katello::deployment_url,
+      ca_key                       => $certs::ca_key,
+      ca_cert                      => $certs::ca_cert_stripped,
+      keystore_password            => $::certs::candlepin::keystore_password,
+      truststore_password          => $::certs::candlepin::keystore_password,
+      enable_basic_auth            => false,
+      consumer_system_name_pattern => '.+',
+      adapter_module               => 'org.candlepin.katello.KatelloModule',
+      amq_enable                   => true,
+      amqp_keystore_password       => $::certs::candlepin::keystore_password,
+      amqp_truststore_password     => $::certs::candlepin::keystore_password,
+      amqp_keystore                => $::certs::candlepin::amqp_keystore,
+      amqp_truststore              => $::certs::candlepin::amqp_truststore,
+    }
   }
 
-  class { '::certs::foreman': }
+  if $enable_qpid {
+    include '::certs::qpid'
+
+    Class['certs'] ~>
+    Class['::certs::qpid'] ~>
+    class { '::qpid':
+      ssl                    => true,
+      ssl_cert_db            => $::certs::nss_db_dir,
+      ssl_cert_password_file => $::certs::qpid::nss_db_password_file,
+      ssl_cert_name          => 'broker',
+      interface              => 'lo',
+    }
+
+    class { '::katello::qpid':
+      client_cert => $certs::qpid::client_cert,
+      client_key  => $certs::qpid::client_key,
+    }
+
+    Class['certs::qpid'] ~> Service['qpidd']
```

Comment:
I wonder why this is needed since there's a ```Class['::certs::qpid'] ~> Class['qpid']``` relation already. I'd expect that to notify the service.